src/main/kotlin/g0601_0700/s0665_non_decreasing_array/Solution.kt

```diff
@@ -0,0 +1,37 @@
+package g0601_0700.s0665_non_decreasing_array
+
+// #Medium #Array #2023_02_14_Time_256_ms_(85.71%)_Space_39.8_MB_(28.57%)
+
+class Solution {
+    fun checkPossibility(nums: IntArray): Boolean {
+        val n = nums.size
+        if (n <= 2) {
+            return true
+        }
+
+        var isModified = false
+        for (i in 0..(n - 2)) {
+            if (nums[i] <= nums[i + 1]) {
+                continue
+            }
+            if (isModified) {
+                return false
+            }
+
+            when {
+                i == 0 || nums[i - 1] <= nums[i + 1] -> {
+                    nums[i] = nums[i + 1]
+                    isModified = true
+                }
+                i == n - 2 || nums[i] <= nums[i + 2] -> {
```

Comment:
Please add more tests to cover codes https://sonarcloud.io/component_measures?metric=new_coverage&selected=javadev_LeetCode-in-Kotlin%3Asrc%2Fmain%2Fkotlin%2Fg0601_0700%2Fs0665_non_decreasing_array%2FSolution.kt&view=list&pullRequest=225&id=javadev_LeetCode-in-Kotlin